internal/handler/wallet/delete_test.go

```diff
@@ -38,38 +40,58 @@ func Test_walletDelete_Handle(t *testing.T) {
 	})
 
 	t.Run("invalid password", func(t *testing.T) {
-		resp := deleteWallet(t, api, nickname)
-
-		verifyStatusCode(t, resp, http.StatusNoContent)
+		// Send password to prompter app and wait for result
+		go func(res chan walletapp.EventData) {
+			prompterApp.App().PromptInput <- "invalid password"
+			// forward test result to test goroutine
+			failRes := <-resChan
 
-		prompterApp.App().PromptInput <- "invalid password"
+			checkResultChannel(t, failRes, false, utils.WrongPassword)
 
-		result := <-resChan
+			// Send cancel to prompter app to unlock the handler
+			prompterApp.App().CtrlChan <- walletapp.Cancel
+		}(testResult)
 
-		checkResultChannel(t, result, false, utils.WrongPassword)
+		resp := deleteWallet(t, api, nickname)
+		verifyStatusCode(t, resp, http.StatusUnauthorized)
 	})
 
 	t.Run("canceled by user", func(t *testing.T) {
-		resp := deleteWallet(t, api, nickname)
+		go func() {
+			// Send wrong password to prompter app and wait for result
+			prompterApp.App().PromptInput <- "this is not the password"
+			// forward test result to test goroutine
+			failRes := <-resChan
 
-		verifyStatusCode(t, resp, http.StatusNoContent)
+			checkResultChannel(t, failRes, false, utils.WrongPassword)
+
+			// Send cancel to prompter app to unlock the handler
+			prompterApp.App().CtrlChan <- walletapp.Cancel
+		}()
+
+		resp := deleteWallet(t, api, nickname)
 
-		prompterApp.App().CtrlChan <- walletapp.Cancel
+		verifyStatusCode(t, resp, http.StatusUnauthorized)
 
 		_, err = wallet.Load(nickname)
 		assert.NoError(t, err)
 	})
 
 	t.Run("delete success", func(t *testing.T) {
+		// Send password to prompter app and wait for result
+		go func(res chan walletapp.EventData) {
+			prompterApp.App().PromptInput <- password
+			// forward test result to test goroutine
+			res <- (<-resChan)
+		}(testResult)
```

Comment:
no need to pass testResult in parameter here. Closures are allowed in go!
`testResult <- (<-resChan)`